### ISSUE TITLE
request_token to Google failing due to missing refresh_token in response

### ISSUE THREAD

**Issue by jgallagher** [programble/inth-oauth2#9]
I'm getting this error when trying to call `Client::request_token` with a `Google` provider:

```
Parse(
    ExpectedFieldType(
        "refresh_token",
        "string"
    )
)
```

It looks like from [Google's docs](https://developers.google.com/identity/protocols/OAuth2WebServer#handlingresponse) that the `refresh_token` isn't supplied by default:

> A token that may be used to obtain a new access token. Refresh tokens are valid until the user revokes access. This field is only present if access_type=offline is included in the authorization code request.

**Comment by programble** [comment]
Strange, this doesn't happen when I run `examples/google.rs`. Can you paste the rest of the flow you're using?

**Comment by jgallagher** [comment]
Hmm, maybe I'm confused about `redirect_uri` and `scope`. It looks like in `examples/google.rs` the scope and redirect URI may be backwards from what I'm doing? This is more or less my code, running inside of Iron:

```rust
const OAUTH_CLIENT_ID: &'static str = "MY_CLIENT_ID";
const OAUTH_CLIENT_SECRET: &'static str = "MY_SECRET";
const OAUTH_CLIENT_REDIRECT_URL: &'static str = "http://my-website.com/oauth2callback";

fn oauth_client() -> inth_oauth2::Client<Google> {
    inth_oauth2::Client::<Google>::new(OAUTH_CLIENT_ID.into(),
                                       OAUTH_CLIENT_SECRET.into(),
                                       Some(OAUTH_CLIENT_REDIRECT_URL.into()))
}

// mounted at http://my-website.com/authorize
fn authorize_handler(_: &mut Request) -> IronResult<Response> {
    let client = oauth_client();
    let redirect_url = client.auth_uri(Some("openid email".into()), Some(OAUTH_STATE.into()))
                             .unwrap();
    let redirect = Redirect(Url::parse(&redirect_url).unwrap());
    Ok(Response::with((status::Found, redirect)))
}

// mounted at http://my-website.com/oauth2callback
fn callback_handler(req: &mut Request) -> IronResult<Response> {
    let client = oauth_client();
    let code: String = // ... get code from req ...
    let token = match client.request_token(&Client::default(), &code) {
        Ok(token) => (),
        Err(e) => {
            // failing here and printing the error above
            println!("err = {:#?}", e);
        }
    };

    Ok(Response::with((status::Ok, "Hi from OAuth callback")))
}
```

**Comment by programble** [comment]
Seems like I was testing with the "Other" client type, which gets issued refresh tokens, rather than the "Web application" client type.

Will need to have two `Provider` implementations for Google.

**Comment by programble** [comment]
Your sample also made me realize that `Client::auth_uri` should be returning a `Uri` and not a `String`.

**Comment by programble** [comment]
Fix in `0.8.0`.
